armory/metrics/task.py

```diff
@@ -1311,6 +1311,210 @@ def carla_od_AP_per_class(y_list, y_pred_list, iou_threshold=0.5, mean=True):
     )
 
 
+def _object_detection_poisoning_get_targeted_mr_dr(
+    y_list, y_pred_list, source_class=None, target_class=None, iou_threshold=0.5, score_threshold=0.5,
+):
+    """ 
+    Modeled after _object_detection_get_tpr_mr_dr_hr.
+    This function computes the targeted misclassification rate and disappearance rate, 
+    for object detection poisoning.  Works for both Regional and Global Misclassification.
+    Object Generation (hallucination) is computed in a separate function 
+    (object_detection_poisoning_targeted_generation_rate) because it has slightly different inputs.
+
+    For Disappearance and Regional Misclassification, it is assumed that 
+    all source class boxes have been triggered.
+    For Disappearance, target_class is None.
+    For Global Misclassification, source_class is None.
+
+    y_list (list): of length equal to the number of input examples. Each element in the list
+        should be a dict with "labels" and "boxes" keys mapping to a numpy array of
+        shape (N,) and (N, 4) respectively where N = number of boxes.
+    y_pred_list (list): of length equal to the number of input examples. Each element in the
+        list should be a dict with "labels", "boxes", and "scores" keys mapping to a numpy
+        array of shape (N,), (N, 4), and (N,) respectively where N = number of boxes.
+
+    returns: a tuple of length 2 (MR, DR) where each element is a list of length equal
+        to the number of images.
+    """
+
+    targeted_misclassification_rate_per_img = []
+    disappearance_rate_per_img = []
+
+
+    for img_idx, (y, y_pred) in enumerate(zip(y_list, y_pred_list)):
+        if source_class:
```

Comment:
I think this maybe needs to be changed to `if source_class is not None`, bc if `source_class` is 0, then the else block is reached, which I believe isn't the desired behavior. I've set a breakpoint() and noticed this happening, and so `gt_labels` includes other label ids:

```
(Pdb) source_class
0
(Pdb) gt_labels
array([1, 1, 1, 1, 1, 1, 1, 1, 2, 2])

```